core/src/main/java/com/linecorp/armeria/common/HttpRequestBuilder.java

```diff
@@ -0,0 +1,397 @@
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType contentType, byte[] content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = HttpData.wrap(content);
+        return this;
+    }
+
+    /**
+     * Sets the content for this request.
+     */
+    public HttpRequestBuilder content(MediaType contentType, HttpData content) {
+        requireNonNull(contentType, "contentType");
+        requireNonNull(content, "content");
+
+        requestHeadersBuilder.contentType(contentType);
+        this.content = content;
+        return this;
+    }
+
+    /**
+     * Sets a header for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .header("authorization", "foo")
+     *            .build();
+     * }</pre>
+     */
+    public HttpRequestBuilder header(CharSequence name, Object value) {
+        requestHeadersBuilder.setObject(requireNonNull(name, "name"), requireNonNull(value, "value"));
+        return this;
+    }
+
+    /**
+     * Sets multiple headers for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/")
+     *            .headers(HttpHeaders.of("authorization", "foo", "bar", "baz"))
+     *            .build();
+     * }</pre>
+     * @see HttpHeaders
+     */
+    public HttpRequestBuilder headers(HttpHeaders httpHeaders) {
+        requireNonNull(httpHeaders, "httpHeaders");
+        httpHeaders.forEach((BiConsumer<AsciiString, String>) requestHeadersBuilder::set);
+        return this;
+    }
+
+    /**
+     * Sets HTTP trailers for this request.
+     */
+    public HttpRequestBuilder trailers(HttpHeaders httpTrailers) {
+        requireNonNull(httpTrailers, "httpTrailers");
+        httpTrailers.forEach((BiConsumer<AsciiString, String>) httpTrailersBuilder::set);
+        return this;
+    }
+
+    /**
+     * Sets a path param for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/{foo}")
+     *            .pathParam("foo", "bar")
+     *            .build(); // GET `/bar`
+     * }</pre>
+     */
+    public HttpRequestBuilder pathParam(String name, Object value) {
+        pathParams.put(requireNonNull(name, "name"), requireNonNull(value, "value"));
+        return this;
+    }
+
+    /**
+     * Sets multiple path params for this request. For example:
+     * <pre>{@code
+     * HttpRequest.builder()
+     *            .get("/{foo}/{bar}")
+     *            .pathParams(Map.of("foo", "bar", "bar", "baz"))
+     *            .build(); // GET `/bar/baz`
+     * }</pre>
+     */
+    public HttpRequestBuilder pathParams(Map<String, Object> pathParams) {
```

Comment:
Could we accept `Iterable<? extends Entry<String, ?>>` instead? I think it will also accept `Map<String, ?>`.